### PR TITLE
Use .equals() on constants and literals to prevent NPEs

### PR DESCRIPTION
**Pull request by wborn**
This should replace all such cases where possible so we don't have to spend time on this anymore when creating/reviewing other PRs.

### CODE REVIEW

**Comment by J-N-K** [comment]
IMO the other way is more readable (closer to natural speech) and I don‘t see the benefit if we have null-notations in place.

**Comment by wborn** [comment]
That is true but lots of code isn't annotated and code and annotations can change over time causing NPEs eventually.

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/provider/AbstractResourceBundleProvider.java [hunk 1]
```diff
@@ -306,15 +306,15 @@ protected void putNewPortfolio(Vendor vendor, List<String> portfolio) {
      */
     protected String getParserType(URL url) {
         String fileName = url.getPath();
-        int fileExtesionStartIndex = fileName.lastIndexOf(".") + 1;
-        if (fileExtesionStartIndex == -1) {
+        int fileExtensionStartIndex = fileName.lastIndexOf(".") + 1;
+        if (fileExtensionStartIndex == -1) {
```
Comment (cweitkamp) [inline bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/provider/AbstractResourceBundleProvider.java:310]: This looks like a bug. How can `fileExtensionStartIndex` equals to -1 when we add 1 in the line before?

We should either do this:

```suggestion
        if (fileName.lastIndexOf(".") == -1) {
```

Or add one later in `fileName.substring(fileExtensionStartIndex + 1);

Maybe do it similar to `AbstractFileProvider#getParserType()`.

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/provider/file/AbstractFileProvider.java [hunk 2]
```diff
@@ -317,7 +317,7 @@ private String getParserType(URL url) {
             return Parser.FORMAT_JSON;
         }
         String fileExtension = fileName.substring(fileName.lastIndexOf(".") + 1);
```
Comment (cweitkamp) [inline bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/provider/file/AbstractFileProvider.java:319]: Store `fileName.lastIndexOf(".")` in a local variable earlier and reuse it here.

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/provider/AbstractResourceBundleProvider.java (hunk 1 above)
Comment (wborn) [reply bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/provider/AbstractResourceBundleProvider.java]: I think you're right so it's fixed now. :+1:

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/provider/file/AbstractFileProvider.java (hunk 2 above)
Comment (wborn) [reply bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/provider/file/AbstractFileProvider.java]: Done! And I threw the Elvis operator in the mix. ?:-)

**Comment by wborn** [comment]
It was quite a bit of a grind this whole PR but it was totally self inflicted. :wink:

**Pull request merged**